hazelcast/src/test/java/com/hazelcast/config/AbstractConfigBuilderTest.java

```diff
@@ -0,0 +1,456 @@
+
+    @Test
+    public abstract void testExecutorConfig();
+
+    @Test
+    public abstract void testDurableExecutorConfig();
+
+    @Test
+    public abstract void testScheduledExecutorConfig();
+
+    @Test
+    public abstract void testCardinalityEstimatorConfig();
+
+    @Test(expected = InvalidConfigurationException.class)
+    public abstract void testCardinalityEstimatorConfigWithInvalidMergePolicy();
+
+    @Test
+    public abstract void testPNCounterConfig();
+
+    @Test
+    public abstract void testMultiMapConfig();
+
+    @Test
+    public abstract void testReplicatedMapConfig();
+
+    @Test
+    public abstract void testListConfig();
+
+    @Test
+    public abstract void testSetConfig();
+
+    @Test
+    public abstract void testMapConfig();
+
+    @Test
+    public abstract void testIndexesConfig();
+
+    @Test
+    public abstract void testAttributeConfig();
+
+    @Test(expected = IllegalArgumentException.class)
+    public abstract void testAttributeConfig_noName_emptyTag();
+
+    @Test(expected = IllegalArgumentException.class)
+    public abstract void testAttributeConfig_noName_singleTag();
+
+    @Test(expected = IllegalArgumentException.class)
+    public abstract void testAttributeConfig_noExtractor();
+
+    @Test(expected = IllegalArgumentException.class)
+    public abstract void testAttributeConfig_emptyExtractor();
+
+    @Test
+    public abstract void testQueryCacheFullConfig();
+
+    @Test
+    public abstract void testMapQueryCachePredicate();
+
+    @Test
+    public abstract void testLiteMemberConfig();
+
+    @Test
+    public abstract void testNonLiteMemberConfig();
+
+    @Test(expected = InvalidConfigurationException.class)
+    public abstract void testNonLiteMemberConfigWithoutEnabledField();
+
+    @Test(expected = InvalidConfigurationException.class)
+    public abstract void testInvalidLiteMemberConfig();
+
+    @Test(expected = InvalidConfigurationException.class)
+    public abstract void testDuplicateLiteMemberConfig();
+
+    @Test
+    public abstract void testMapNativeMaxSizePolicy();
+
+    @Test
+    public abstract void testInstanceName();
+
+    @Test
+    public abstract void testUserCodeDeployment();
+
+    @Test
+    public abstract void testCRDTReplicationConfig();
+
+    @Test
+    public abstract void testGlobalSerializer();
+
+    @Test
+    public abstract void testJavaSerializationFilter();
+
+    @Test
+    public abstract void testHotRestart();
+
+    @Test
+    public abstract void testMapEvictionPolicyClassName();
+
+    @Test
+    public abstract void testMapEvictionPolicyIsSelected_whenEvictionPolicySet();
+
+    @Test
+    public abstract void testCachePermission();
+
+    @Test
+    public abstract void testConfigPermission();
+
+    @Test
+    public abstract void testAllPermissionsCovered();
+
+    // TODO what is this really testing?
+    // The thrown Exception has nothing to do with the eviction config,
+    // it blames NATIVE in-memory-format in OS build...
```

Comment:
If unsure if needed, maybe remove it in this PR?

Reply:
I split this test case into two. One to check we're failing fast with invalid eviction config, and another that verifies we also fail fast with `NATIVE` cache `in-memory-format` in the OSS.